homeassistant/components/uptimerobot/__init__.py

```diff
@@ -1 +1,85 @@
-"""The uptimerobot component."""
+"""The Uptime Robot integration."""
+from __future__ import annotations
+
+import async_timeout
+from pyuptimerobot import UptimeRobot
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_PLATFORM
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    API_ATTR_MONITORS,
+    API_ATTR_OK,
+    API_ATTR_STAT,
+    CONNECTION_ERROR,
+    COORDINATOR_UPDATE_INTERVAL,
+    DOMAIN,
+    LOGGER,
+    PLATFORMS,
+    MonitorData,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Uptime Robot integration."""
+    # Iterate all entries for binart_sensor to only get uptimerobot
+    hass.data.setdefault(DOMAIN, {})
+    for entry in config.get(BINARY_SENSOR, []):
+        if entry[CONF_PLATFORM] == DOMAIN:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
```

Comment:
It's safer to do this from the platform legacy setup function.

Reply:
Do you have any examples of integrations that have done that recently?

Reply:
https://github.com/home-assistant/core/blob/1f9331f9dbf88c2ded607a4c977d4da122ec20ce/homeassistant/components/xiaomi_miio/fan.py#L414-L426

Reply:
It's creating a task, but other than that there is no difference? both are called with `async_setup_platform`

Reply:
nevermind!